ocaml/driver/compile_common.ml

```diff
@@ -123,7 +129,13 @@ let parse_impl i =
 
 let typecheck_impl i parsetree =
   parsetree
-  |> Profile.(record typing)
+  |> Profile.(
+    record_with_counters
+      ~counter_f:(fun (typed_tree : Typedtree.implementation) ->
+        Profile_counters_functions.(
+          count_language_extensions
+            (Typedtree_implementation_output typed_tree)))
+      typing)
```

Comment:
(Maybe factor out with lines 63-69,
if possible / easy.)

Reply:
It seems slightly less readable factored out as the two use different variant constructors (not treated as functions in OCaml) and one uses `Profile.record_with_counters` whereas the other uses `Profile.record_call_with_counters`.